### PR TITLE
PP-10940 Set a connection manager for webhook message sender client

### PR DESCRIPTION
**Pull request by sfount**
Use a thread safe pooling connection manager the http client used to send webhook messages. This should be configured to allow at least the number of webhook message sending threads number of connections to lease.

### CODE REVIEW

File: src/main/java/uk/gov/pay/webhooks/app/WebhookMessageSendingQueueProcessorConfig.java [hunk 1]
```diff
@@ -29,14 +33,18 @@ public int getThreadDelayInMilliseconds() {
...
+    public int getWebhookMessageSendingQueueProcessorNumberOfThreads() {
+        return webhookMessageSendingQueueProcessorNumberOfThreads;
     }
 
     public int getInitialDelayInMilliseconds() {
         return initialDelayInMilliseconds;
     }
 
+    public int getConnectionPoolNumberOfThreads() {
+        return connectionPoolNumberOfThreads;
```
Comment (sfount) [inline src/main/java/uk/gov/pay/webhooks/app/WebhookMessageSendingQueueProcessorConfig.java:45]: I think calling these "threads" though isn't quite right, it's essentially the maximum number of simultaneous http connections that can be available in the pool at any given time for a client to use, the client can access those connections from any thread.

I think something like "httpClientConnectionPoolSize" or something like that might better reflect it

**Review by sfount — COMMENTED** [review]
Implementation looks great, question about the naming

File: src/main/java/uk/gov/pay/webhooks/app/WebhookMessageSendingQueueProcessorConfig.java (hunk 1 above)
Comment (stephencdaly) [reply src/main/java/uk/gov/pay/webhooks/app/WebhookMessageSendingQueueProcessorConfig.java]: Updated

**Review by sfount — COMMENTED** [review]
✅

**Review by stephencdaly — APPROVED** [review]

**Pull request merged**